pilot/pkg/model/push_context.go

```diff
@@ -985,6 +1004,33 @@ func (ps *PushContext) initDestinationRules(env *Environment) error {
 	return nil
 }
 
+// AuthenticationPolicyForWorkload returns the matching auth policy for a given service
```

Comment:
this function checks `AuthNPolicies` map has the entry, if no return nil.

And the map is only initialized once? Who is responsible for keeping the map in sync with the `IstioConfigStore` and invalidate the cache?

@diemtvu @GregHanson

Reply:
@incfly are you implying that AuthNPolicies work different than the DestinationRules, VirtualServices, etc in push_context?  I thought the context is cached and created during a push, and then the process repeats during the next push?

Reply:
Got it. It's due to my own lack of the familarity. We init the push policy at the beginning for the push, for many other crds as well.

The outcome is that we won't exercise `AuthenticationPolicyForWorkload` from config store by iterating & matching multiple times. Sounds good to me.

Reply:
Resolving.